x-pack/filebeat/input/awss3/metrics.go

```diff
@@ -147,13 +145,12 @@ func (m *inputMetrics) updateSqsWorkerUtilization() {
 	m.sqsWorkerUtilizationLastUpdate = now
 }
 
-func newInputMetrics(id string, optionalParent *monitoring.Registry, maxWorkers int) *inputMetrics {
-	reg, unreg := inputmon.NewInputRegistry(inputName, id, optionalParent)
+func newInputMetrics(v2ctx v2.Context, maxWorkers int) *inputMetrics {
```

Comment:
Since the common pattern with these `newInputMetrics` functions is that they don't use any context field except the registry, and the tests end up having to wrap all the registries in otherwise empty contexts in many locations, I think these should just take a registry instead of a context, so there's no uncertainty about what they depend on and less potentially-misleading test boilerplate

Reply:
It's also a possibility. I started passing the context, then for consistency I kept the pattern. I could change, but then, ideally, to keep them consistent, it's mean to change it on all inputs. I don't think it's worth the effort. Also passing the context creates the association between the input metrics and the v2.Context.